cibuildwheel/bashlex_eval.py

```diff
@@ -37,11 +41,11 @@ def evaluate_node(node, context):
         raise ValueError('Unsupported bash construct: "%s"' % node.word)
 
 
-def evaluate_word_node(node, context):
+def evaluate_word_node(node: bashlex.ast.node, context: NodeExecutionContext) -> str:
     word_start = node.pos[0]
     word_end = node.pos[1]
     word_string = context.input[word_start:word_end]
-    letters = list(word_string)
+    letters = list(word_string)  # type: List[Optional[str]]
```

Comment:
why optional?

Reply:
Because I got an error that `None` is assigned further down. If you remove the annotation, it should show up.

Reply:
We could set them to `''` empty string instead of None, and make it non-optional.

Reply:
If you prefer? It does clean up `''.join(l for l in letters if l is not None)` to just `''.join(letters)` as well.